### ISSUE TITLE
ressource openstack_compute_interface_attach_v2: Import does not import instance_id or is not refreshed correctly

### ISSUE THREAD

**Issue by buec011** [terraform-providers/terraform-provider-openstack#554]
### Terraform Version
Terraform v0.11.11
provider.openstack: version = "~> 1.13"

### Affected Resource(s)
Please list the resources as a list, for example:
- openstack_compute_interface_attach_v2

### Terraform Configuration Files
```hcl
resource "openstack_compute_instance_v2" "basic" {
  name            = "basic"
  image_id        = "3f09d0ec-ddd7-47de-9c05-d74a91a1c15c"
  flavor_id       = "3"

  network {
    name = "test-network"
  }
}

resource openstack_networking_port_v2 port {
   network_id="48a2e109-02a9-4a8e-aa4b-ccc4018f02e5"
}

resource openstack_compute_interface_attach_v2 port {
  instance_id = "${openstack_compute_instance_v2.basic.id}"
  port_id  = "${openstack_networking_port_v2.port.port_id}"
}
output port_id {
   value="${openstack_networking_port_v2.port.id}"
}
output instance_id {
   value="${openstack_compute_instance_v2.basic.id}"
}
```

### Expected Behavior
If we Import an port attachment, terraform plan / apply should not attempt to change anything

### Actual Behavior
terraform apply wants to destroy and re-create the attachment ressource. If you add instance_id manually to the state file, everything is ok.

```
-/+ openstack_compute_interface_attach_v2.port (new resource required)
      id:          "06899a02-d7f9-4899-b37f-726110682643/58a44d9f-de18-4e33-bf25-c62d50a9bde8" => <computed> (forces new resource)
      instance_id: "" => "06899a02-d7f9-4899-b37f-726110682643" (forces new resource)
      network_id:  "48a2e109-02a9-4a8e-aa4b-ccc4018f02e5" => <computed>
      port_id:     "58a44d9f-de18-4e33-bf25-c62d50a9bde8" => "58a44d9f-de18-4e33-bf25-c62d50a9bde8"
      region:      "nova" => <computed>

```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply` (rollout mentioned template)
2. `terraform state rm openstack_compute_interface_attach_v2.port` ("Forget" the attachment, we want to import it)
3. `terraform import openstack_compute_interface_attach_v2.port 06899a02-d7f9-4899-b37f-726110682643/58a44d9f-de18-4e33-bf25-c62d50a9bde8`
4. `terraform plan`  ---> Wants to change attachment

**Comment by jtopjian** [comment]
@buec011 Thank you for reporting this - this is indeed incorrect behavior.

Internal notes: we need to add an import test for this resource. That should easily reproduce this issue. 

The root cause is not doing a `d.Set("instance_id", ...)` in the Read function: https://github.com/terraform-providers/terraform-provider-openstack/blob/master/openstack/resource_openstack_compute_interface_attach_v2.go#L132

**Comment by buec011** [comment]
If I create a port and attach it, terraform has the following entry.
```
                    "depends_on": [
                        "openstack_compute_instance_v2.basic",
                        "openstack_networking_port_v2.port"
                    ],
```
If I import a port, and run terraform refresh/plan/apply, this is not set. Does it matter?

**Comment by buec011** [comment]
PS: Great work! :-)

**Comment by jtopjian** [comment]
Are you seeing that `depends_on` entry in the Terraform state? If so, it's probably just internal to the state. Since you are referencing the resources like this:

```hcl
resource openstack_compute_interface_attach_v2 port {
  instance_id = "${openstack_compute_instance_v2.basic.id}"
  port_id  = "${openstack_networking_port_v2.port.id}"
}
```

The `${ ... }` entries are creating implied `depends_on`.

**Comment by buec011** [comment]
> Are you seeing that `depends_on` entry in the Terraform state?

Yes, I am referring to the terraform state file. I just worry if an import or refresh need to set those depend_on in the state file?

**Comment by jtopjian** [comment]
No, I don't think so. I think if you create a `.tf` file that looks like the example above (using the implicit resource references), that should be good enough.
